File: Behavioral/Strategy/JavaScript/creditCardPayment.js
// creditCardPayment.js

const PaymentStrategy = require("./paymentStrategy");

class CreditCardPayment extends PaymentStrategy {
  constructor(cardNumber) {
    super();
    this.cardNumber = cardNumber;
  }

  pay(amount) {
    console.log(`Paid $${amount} using Credit Card (Card Number: ${this.cardNumber})`);
  }
}

module.exports = CreditCardPayment;

File: Behavioral/Strategy/Python/credit_card_payment.py
# credit_card_payment.py

from payment_strategy import PaymentStrategy

class CreditCardPayment(PaymentStrategy):
  def __init__(self, card_number):
    self.card_number = card_number

  def pay(self, amount):
    print(f"Paid ${amount} using Credit Card (Card Number: {self.card_number})")
